airbyte-webapp/src/components/connectorBuilder/Builder/BuilderOneOf.tsx

```diff
@@ -0,0 +1,66 @@
+import { useField } from "formik";
+import React, { useEffect } from "react";
+
+import GroupControls from "components/GroupControls";
+import { ControlLabels } from "components/LabeledControl";
+import { DropDown } from "components/ui/DropDown";
+
+interface Option {
+  label: string;
+  value: string;
+  default?: object;
+}
+
+interface OneOfOption {
+  label: string; // label shown in the dropdown menu
+  typeValue: string; // value to set on the `type` field for this component - should match the oneOf type definition
+  default?: object; // default values for the path
+  children?: React.ReactNode;
+}
+
+interface BuilderOneOfProps {
+  options: OneOfOption[];
+  path: string; // path to the oneOf component in the json schema
+  label: string;
+  tooltip: string;
+}
+
+export const BuilderOneOf: React.FC<BuilderOneOfProps> = ({ options, path, label, tooltip }) => {
+  const [, , oneOfPathHelpers] = useField(path);
+  const typePath = `${path}.type`;
+  const [typePathField, , typePathHelpers] = useField(typePath);
+  const value = typePathField.value ?? options[0].typeValue;
+
+  useEffect(() => {
+    typePathHelpers.setValue(value);
+  }, []); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
I added this at one point but later removed it from BuilderFields since I didn't think it was actually needed anymore. Could you double check if this is necessary to keep?

Reply:
Good catch, we don't and I think it's better to truthfully show no selected option instead of defaulting to the first one as that wouldn't be reflected in the actual form state.